src/main/java/org/openrewrite/gradle/RewriteRecipeOriginPlugin.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.gradle;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.provider.Property;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPomLicense;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.publish.maven.internal.publication.DefaultMavenPom;
+import org.gradle.api.tasks.Copy;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.language.jvm.tasks.ProcessResources;
+
+import java.io.File;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.eclipse.jgit.util.StringUtils.capitalize;
+
+public class RewriteRecipeOriginPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        String recipeLicenseUrl = determineLicenseDetail(MavenPomLicense::getUrl, project);
+        String recipeLicenseName = determineLicenseDetail(MavenPomLicense::getName, project);
+        String repoUrl = determineRepoUrl(project);
+
+        JavaPluginExtension java = project.getExtensions().findByType(JavaPluginExtension.class);
+        if (java == null) {
+            return;
+        }
+        SourceSet mainSource = java.getSourceSets().getByName("main");
+        TaskContainer tasks = project.getTasks();
+        TaskProvider<Copy> copyOrigins = tasks.register("copyOrigins", Copy.class);
+        //noinspection UnstableApiUsage
+        tasks.named("processResources", ProcessResources.class).configure(task -> {
+            task.dependsOn(copyOrigins);
+        });
```

Comment:
`ProcessResources` is itself a `Copy` task so you can configure it rather than creating a new copy task